### PR TITLE
Make VPN affiliate code re-usable for other products (fix #13575)

### LINKED ISSUE #13575: Make VPN affiliate code re-usable for other products
Remove the "VPN" references from the VPN affiliate JS to make it re-usable for other products.

### PR DESCRIPTION
**Pull request by stephaniehobson**
## One-line summary

Make VPN affiliate code re-usable for other products.

## Significant changes and points to review

- moves cookie banner HTML to separate file and makes notification more generic
- re-name and move other associated files to remove VPN references
- rename various variable names to remove VPN references
- add affiliate strings to global FTL files so translations are always available, because this seems like the kind of step that is easy to forget when adding it to a new page.
- update docs

## Issue / Bugzilla link

Fix #13575 
See also #13574

## Testing

http://localhost:8000/fr/products/vpn/pricing/?cjevent=test
`npm run test`

### CODE REVIEW

File: bedrock/settings/base.py [hunk 1]
```diff
@@ -231,6 +231,7 @@ def data_path(*args):
 
 # Global L10n files.
 FLUENT_DEFAULT_FILES = [
+    "affiliate.ftl",
```
Comment (alexgibson) [inline bedrock/settings/base.py:234]: Nit: remove filename suffix for consistency?
```suggestion
    "affiliate",
```

File: docs/attribution/0005-affiliate-marketing.rst [hunk 2]
```diff
@@ -4,17 +4,16 @@
...
+Mozilla CJMS affiliate attribution
+==================================
 
-The affiliate attribution flow for the Mozilla :abbr:`VPN (Virtual Private Network)`
-`landing page`_ comprises an integration between the `Commission Junction (CJ)`_
-affiliate marketing event system, bedrock, and the VPN product team's
-`CJ micro service (CJMS)`_.
+The CJMS affiliate attribution flow comprises an integration between
+the `Commission Junction (CJ)`_ affiliate marketing event system, bedrock,
+and the VPN product team's `CJ micro service (CJMS)`_.
```
Comment (alexgibson) [inline docs/attribution/0005-affiliate-marketing.rst:13]: Should this be `Security and Privacy team's`?

File: media/static-bundles.json.orig [hunk 3]
```diff
@@ -1,1960 +0,0 @@
-{
```
Comment (alexgibson) [inline media/static-bundles.json.orig:1]: I don't see this file in the `main` branch, which is weird. Does this branch need a rebase perhaps?

File: bedrock/base/templates/includes/affiliate-banner.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{#
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#}
+
+<aside class="mzp-c-notification-bar affiliate-notification hide-from-legacy-ie">
+  {% if not affiliate_message %}
+    {% set attrs = 'href="' ~ url('privacy.notices.websites') ~ '" target="_blank" rel="noopener noreferrer"' %}
```
Comment (alexgibson) [inline bedrock/base/templates/includes/affiliate-banner.html:9]: This is left over from the existing banner, but we can remove `rel="noopener noreferrer"` for same-site links?

File: bedrock/products/templates/products/vpn/landing-refresh.html [hunk 5]
```diff
@@ -48,7 +48,7 @@
 {% block content %}
 <main>
   {% if vpn_affiliate_attribution_enabled %}
-    {% include 'products/vpn/includes/cookie-notification.html' %}
```
Comment (alexgibson) [inline bedrock/products/templates/products/vpn/landing-refresh.html:51]: Can we remove `products/vpn/includes/cookie-notification.html`?

**Review by alexgibson — CHANGES_REQUESTED** [review]
A couple of minor things, but otherwise this all looks good to me. I'd suggest giving this branch a rebase, as it seems a bit behind (and so you can test it with the latest protocol update?)

r+wc

**Comment by stephaniehobson** [comment]
Rebased and updated.

File: l10n/en/affiliate.ftl [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/en-US/products/vpn/pricing/?cjevent=test
+
+## CJMS Affiliate cookie notice
+
+# Variables:
+#   $attrs (string) - link to https://www.mozilla.org/en-US/privacy/websites/ with additional attributes.
```
Comment (alexgibson) [inline l10n/en/affiliate.ftl:10]: Nit: remove locale string
```suggestion
#   $attrs (string) - link to https://www.mozilla.org/privacy/websites/ with additional attributes.
```

File: l10n/en/affiliate.ftl [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/en-US/products/vpn/pricing/?cjevent=test
```
Comment (alexgibson) [inline l10n/en/affiliate.ftl:5]: NIt remove locale string
```suggestion
### URL: https://www-dev.allizom.org/products/vpn/pricing/?cjevent=test
```

**Review by alexgibson — APPROVED** [review]
r+wc

**Comment by stephaniehobson** [comment]
This is ready to merge but can wait until after All Hands :)

**Pull request merged**